# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1552671441 25200
#      Fri Mar 15 10:37:21 2019 -0700
# Node ID c81b471cdb28075fc73eb3eca5bf92e4f7880f19
# Parent  072d0610cc5634b427f673d5032f30f8f49622dd
build: avoid test syntax error in sundials configuration (bug #55926)

* acinclude.m4 (OCTAVE_CHECK_SUNDIALS_SUNLINSOL_KLU): Quote use of
$octave_cv_sundials_sunlinsol_klu in test command and swap order, as
it may not have been set.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2264,18 +2264,18 @@ AC_DEFUN([OCTAVE_CHECK_SUNDIALS_SUNLINSO
          #include <sunlinsol/sunlinsol_klu.h>
          #endif
          ]], [[
          SUNKLU (0, 0);
       ]])],
       octave_cv_sundials_sunlinsol_klu=yes,
       octave_cv_sundials_sunlinsol_klu=no)
     ])])
-  if test $octave_cv_sundials_sunlinsol_klu = yes \
-     && test "x$ac_cv_header_sunlinsol_sunlinsol_klu_h" = xyes; then
+  if test "x$ac_cv_header_sunlinsol_sunlinsol_klu_h" = xyes \
+     && test "x$octave_cv_sundials_sunlinsol_klu" = xyes; then
     AC_DEFINE(HAVE_SUNDIALS_SUNLINSOL_KLU, 1,
       [Define to 1 if SUNDIALS IDA is configured with SUNLINSOL_KLU enabled.])
   else
     warn_sundials_sunlinsol_klu="SUNDIALS IDA library not configured with SUNLINSOL_KLU or sunlinksol_klu.h is not usable; ode15i and ode15s will not support the sparse Jacobian feature"
     OCTAVE_CONFIGURE_WARNING([warn_sundials_sunlinsol_klu])
   fi
 ])
 dnl
